Adds Deprecation warnings for ctype solar-x/solar-y

tanishy7777 opened the pull request:
## PR Description
Adds deprecation warnings for ctype solar-x/solar-y in `mapbase.py`. When pytest is run the user is shown with deprecation warnings. 

Fixes #5569

tanishy7777 commented:
https://github.com/sunpy/sunpy/pull/6901#issuecomment-1501906240
@nabobalis Could you please elaborate what you mean by this.

nabobalis commented:
> Due to the Deprecation warnings the tests seem to be failing right now. I understand in the previous PR a similar problem was faced to which you had responded with [#6901 (comment)](https://github.com/sunpy/sunpy/pull/6901#issuecomment-1501906240) @nabobalis Could you please elaborate what you mean by this.

We have to stop this warning from being raised in the map sources that require `("solar-x", "solar_x")`.

nabobalis commented on sunpy/map/sources/soho.py:
```diff
@@ -37,6 +39,22 @@ class EITMap(GenericMap):
     def __init__(self, data, header, **kwargs):
         super().__init__(data, header, **kwargs)
 
+        ctype1 = self.meta.get('ctype1', None)
+        if not ctype1:
+            warn_metadata("Missing CTYPE1 from metadata, assuming CTYPE1 is HPLN-TAN")
+            ctype1 = 'HPLN-TAN'
+
+        ctype2 = self.meta.get('ctype2', None)
+        if not ctype2:
+            warn_metadata("Missing CTYPE2 from metadata, assuming CTYPE2 is HPLT-TAN")
+            ctype2 = 'HPLT-TAN'
+
+        if ctype1.lower() in ("solar-x", "solar_x"):
+            ctype1 = 'HPLN-TAN'
+
+        if ctype2.lower() in ("solar-y", "solar_y"):
+            ctype2 = 'HPLT-TAN'
```
This should not raise the warning since this is how these sources deal with this. 

This should override the main method in mapbase

tanishy7777 commented:
> We have to stop this warning from being raised in the map sources that require `("solar-x", "solar_x")`.

ok got it. So as per the original issue #5569 and your comments. I have moved the if block inside the `map source` folder files wherever the old functionality of modiftying ctype was being used. Also I have removed the deprecation warnings. Please let me know in case any more changes are to be done.

nabobalis commented:
> > We have to stop this warning from being raised in the map sources that require `("solar-x", "solar_x")`.
> 
> ok got it. So as per the original issue #5569 and your comments. I have moved the if block inside the `map source` folder files wherever the old functionality of modiftying ctype was being used. Also I have removed the deprecation warnings. Please let me know in case any more changes are to be done. @nabobalis

Unofrutaluy, these changes as they stand are incorrect.

nabobalis commented on sunpy/map/mapbase.py:
```diff
@@ -1228,15 +1228,6 @@ def coordinate_system(self):
             warn_metadata("Missing CTYPE2 from metadata, assuming CTYPE2 is HPLT-TAN")
             ctype2 = 'HPLT-TAN'
 
-        # Astropy WCS does not understand the SOHO default of "solar-x" and
-        # "solar-y" ctypes.  This overrides the default assignment and
-        # changes it to a ctype that is understood.  See Thompson, 2006, A.&A.,
-        # 449, 791.
-        if ctype1.lower() in ("solar-x", "solar_x"):
-            ctype1 = 'HPLN-TAN'
-
-        if ctype2.lower() in ("solar-y", "solar_y"):
-            ctype2 = 'HPLT-TAN'
```
The goal is to raise the warning in mapbase with the future aim of removing this code.

Then the map sources which require these logic blocks will overload this method to not raise the warning anymore.

tanishy7777 replied:
> The goal is to raise the warning in mapbase with the future aim of removing this code.
> 
> Then the map sources which require these logic blocks will overload this method to not raise the warning anymore and check for solar-x and solar-y

oh ok that makes sense thx

tanishy7777 commented:
@nabobalis  I have made the changes. Is it fine now?

nabobalis commented on sunpy/map/mapbase.py:
```diff
@@ -1232,13 +1232,27 @@ def coordinate_system(self):
         # "solar-y" ctypes.  This overrides the default assignment and
         # changes it to a ctype that is understood.  See Thompson, 2006, A.&A.,
         # 449, 791.
+
+        # Check for deprecated or alternate values
+        ctype1, ctype2 = self._handle_deprecated_ctypes(ctype1, ctype2)
+
+        return SpatialPair(ctype1, ctype2)
```
```suggestion
        # Check for deprecated or alternate values
        ctype1, ctype2 = self._handle_deprecated_ctypes(ctype1, ctype2)
        return SpatialPair(ctype1, ctype2)
```

nabobalis commented on sunpy/map/mapbase.py:
```diff
@@ -1232,13 +1232,27 @@ def coordinate_system(self):
+    
+    def _handle_deprecated_ctypes(self, ctype1, ctype2):
+        """
+        Handle deprecated or alternate CTYPE values.
```
```suggestion
        Handle deprecated CTYPE values.
```

nabobalis commented on sunpy/map/sources/soho.py:
```diff
@@ -33,15 +34,31 @@ class EITMap(GenericMap):
     * `SOHO EIT User Guide <https://umbra.nascom.nasa.gov/eit/eit_guide/>`_
 
     """
+    def _handle_deprecated_ctypes(self, ctype1, ctype2):
```
This needs to override the `coordinate_system` method on the GenericMap class instead.

tanishy7777 replied:
ok got it

nabobalis commented on sunpy/map/sources/soho.py:
```diff
@@ -33,15 +34,31 @@ class EITMap(GenericMap):
     def __init__(self, data, header, **kwargs):
         super().__init__(data, header, **kwargs)
 
+        ctype1 = self.meta.get('ctype1', None)
+        ctype2 = self.meta.get('ctype2', None)
+
+        self._handle_deprecated_ctypes(ctype1, ctype2)
```
The changes we want should not modify the init of any of the map sources.

We need to check if that approach works.

tanishy7777 replied:
yep got it

tanishy7777 commented:
@nabobalis I have overridden the coordinate_system method as you suggested. Please have a look at it now

nabobalis commented on sunpy/map/mapbase.py:
```diff
@@ -1232,13 +1232,18 @@ def coordinate_system(self):
             ctype2 = 'HPLT-TAN'
 
         return SpatialPair(ctype1, ctype2)
+    
```
```suggestion
```

nabobalis commented on sunpy/map/mapbase.py:
```diff
@@ -1232,13 +1232,18 @@ def coordinate_system(self):
         # "solar-y" ctypes.  This overrides the default assignment and
         # changes it to a ctype that is understood.  See Thompson, 2006, A.&A.,
         # 449, 791.
+
+        # Check for deprecated or alternate values
```
```suggestion
```

nabobalis commented on sunpy/map/sources/hinode.py:
```diff
@@ -48,6 +50,7 @@ class XRTMap(GenericMap):
     filter_wheel2_measurements = ["Open", "Al_mesh", "Al_thick",
                                   "Be_thick", "Gband", "Ti_poly"]
 
+
```
```suggestion
```

nabobalis commented on sunpy/map/sources/hinode.py:
```diff
@@ -56,9 +59,26 @@ def __init__(self, data, header, **kwargs):
         if fw2.lower() not in _lower_list(self.filter_wheel2_measurements):
             raise ValueError(f'Unexpected filter wheel 2 {fw2} in header.')
         super().__init__(data, header, **kwargs)
+
```
```suggestion
```

nabobalis commented on sunpy/map/sources/hinode.py:
```diff
@@ -56,9 +59,26 @@ def __init__(self, data, header, **kwargs):
+        """
+        Override the default implementation to handle SOTMap-specific logic for CTYPE values.
+        """
+        ctype1 = self.meta.get('ctype1', None)
+        ctype2 = self.meta.get('ctype2', None)
+
+        if ctype1.lower() in ("solar-x", "solar_x"):
+            ctype1 = 'HPLN-TAN'
+
+        if ctype2.lower() in ("solar-y", "solar_y"):
+            ctype2 = 'HPLT-TAN'
+
+        return SpatialPair(ctype1, ctype2)
```
```suggestion
        ctype1 = self.meta.get('ctype1', None)
        ctype2 = self.meta.get('ctype2', None)
        if ctype1.lower() in ("solar-x", "solar_x"):
            ctype1 = 'HPLN-TAN'
        if ctype2.lower() in ("solar-y", "solar_y"):
            ctype2 = 'HPLT-TAN'
        return SpatialPair(ctype1, ctype2)
```

nabobalis commented on sunpy/map/sources/hinode.py:
```diff
@@ -168,6 +188,22 @@ def __init__(self, data, header, **kwargs):
+        """
+        Override the default implementation to handle SOTMap-specific logic for CTYPE values.
+        """
+        ctype1 = self.meta.get('ctype1', None)
+        ctype2 = self.meta.get('ctype2', None)
+
+        if ctype1.lower() in ("solar-x", "solar_x"):
+            ctype1 = 'HPLN-TAN'
+
+        if ctype2.lower() in ("solar-y", "solar_y"):
+            ctype2 = 'HPLT-TAN'
+
+        return SpatialPair(ctype1, ctype2)
```
```suggestion
        ctype1 = self.meta.get('ctype1', None)
        ctype2 = self.meta.get('ctype2', None)
        if ctype1.lower() in ("solar-x", "solar_x"):
            ctype1 = 'HPLN-TAN'
        if ctype2.lower() in ("solar-y", "solar_y"):
            ctype2 = 'HPLT-TAN'
        return SpatialPair(ctype1, ctype2)
```

nabobalis commented on sunpy/map/sources/soho.py:
```diff
@@ -12,6 +12,7 @@
 from sunpy.map.sources.source_type import source_stretch
 from sunpy.time import parse_time
 
+
```
```suggestion
```

nabobalis commented on sunpy/map/sources/soho.py:
```diff
@@ -41,7 +42,23 @@ def __init__(self, data, header, **kwargs):
+        """
+        Override the default implementation of coordinate_system to handle EITMAP-specific logic for CTYPE values.
+        """
+        ctype1 = self.meta.get('ctype1', None)
+        ctype2 = self.meta.get('ctype2', None)
 
+        if ctype1.lower() in ("solar-x", "solar_x"):
+            ctype1 = 'HPLN-TAN'
+
+        if ctype2.lower() in ("solar-y", "solar_y"):
+            ctype2 = 'HPLT-TAN'
+
+        return SpatialPair(ctype1, ctype2)
```
```suggestion
        ctype1 = self.meta.get('ctype1', None)
        ctype2 = self.meta.get('ctype2', None)
        if ctype1.lower() in ("solar-x", "solar_x"):
            ctype1 = 'HPLN-TAN'
        if ctype2.lower() in ("solar-y", "solar_y"):
            ctype2 = 'HPLT-TAN'
        return SpatialPair(ctype1, ctype2)
```

nabobalis commented on sunpy/map/sources/soho.py:
```diff
@@ -104,14 +121,29 @@ class LASCOMap(GenericMap):
     * `SOHO Mission Page <https://sohowww.nascom.nasa.gov/>`_
-    """
-
+    """    
```
```suggestion
    """
```

nabobalis commented on sunpy/map/sources/soho.py:
```diff
@@ -104,14 +121,29 @@ class LASCOMap(GenericMap):
         self.plot_settings['norm'] = ImageNormalize(
             stretch=source_stretch(self.meta, PowerStretch(0.5)), clip=False)
+        
+    @property
+    def coordinate_system(self):
+        """
+        Override the default implementation to handle LASCOMAP-specific logic for CTYPE values.
+        """
+        ctype1 = self.meta.get('ctype1', None)
+        ctype2 = self.meta.get('ctype2', None)
+
+        if ctype1.lower() in ("solar-x", "solar_x"):
+            ctype1 = 'HPLN-TAN'
+
+        if ctype2.lower() in ("solar-y", "solar_y"):
+            ctype2 = 'HPLT-TAN'
+
+        return SpatialPair(ctype1, ctype2)
```
```suggestion
    @property
    def coordinate_system(self):
        """
        Override the default implementation to handle LASCOMAP-specific logic for CTYPE values.
        """
        ctype1 = self.meta.get('ctype1', None)
        ctype2 = self.meta.get('ctype2', None)
        if ctype1.lower() in ("solar-x", "solar_x"):
            ctype1 = 'HPLN-TAN'
        if ctype2.lower() in ("solar-y", "solar_y"):
            ctype2 = 'HPLT-TAN'
        return SpatialPair(ctype1, ctype2)
```

nabobalis commented on sunpy/map/sources/trace.py:
```diff
@@ -10,6 +10,7 @@
 from sunpy.map.mapbase import GenericMap, SpatialPair
 from sunpy.map.sources.source_type import source_stretch
 
+
```
```suggestion
```

nabobalis commented on sunpy/map/sources/trace.py:
```diff
@@ -54,6 +55,22 @@ def __init__(self, data, header, **kwargs):
         self.plot_settings['norm'] = ImageNormalize(
             stretch=source_stretch(self.meta, LogStretch()), clip=False)
+        
+    @property
+    def coordinate_system(self):
+        """
+        Override the default implementation to handle TRACEMAP-specific logic for CTYPE values.
+        """
+        ctype1 = self.meta.get('ctype1', None)
+        ctype2 = self.meta.get('ctype2', None)
+
+        if ctype1.lower() in ("solar-x", "solar_x"):
+            ctype1 = 'HPLN-TAN'
+
+        if ctype2.lower() in ("solar-y", "solar_y"):
+            ctype2 = 'HPLT-TAN'
+
+        return SpatialPair(ctype1, ctype2)
```
```suggestion
    @property
    def coordinate_system(self):
        """
        Override the default implementation to handle TRACEMAP-specific logic for CTYPE values.
        """
        ctype1 = self.meta.get('ctype1', None)
        ctype2 = self.meta.get('ctype2', None)
        if ctype1.lower() in ("solar-x", "solar_x"):
            ctype1 = 'HPLN-TAN'
        if ctype2.lower() in ("solar-y", "solar_y"):
            ctype2 = 'HPLT-TAN'
        return SpatialPair(ctype1, ctype2)
```

nabobalis commented:
Can we add some unit tests for these sources to make sure nothing is broken?

tanishy7777 commented:
> Can we add some unit tests for these sources to make sure nothing is broken?

Sure will work on it. I was thinking since we have overridden the coordinate_system method and that changes the ctype1 and ctype2 so I will test for those changes. Will that be enough? @nabobalis

nabobalis commented:
> > Can we add some unit tests for these sources to make sure nothing is broken?
> 
> Sure will work on it. I was thinking since we have overridden the coordinate_system method and that changes the ctype1 and ctype2 so I will test for those changes. Will that be enough? @nabobalis

Yeah sounds good.

tanishy7777 commented:
@nabobalis cool I have added them

nabobalis commented on sunpy/map/mapbase.py:
```diff
@@ -1233,9 +1233,11 @@ def coordinate_system(self):
         # changes it to a ctype that is understood.  See Thompson, 2006, A.&A.,
         # 449, 791.
         if ctype1.lower() in ("solar-x", "solar_x"):
+            warn_deprecated("CTYPE1 value 'solar-x'/'solar_x' is deprecated, use 'HPLN-TAN' instead.")
```
Can we add a generic map test which hits these warns and checks the message?

tanishy7777 replied:
Cool will do that

nabobalis commented on sunpy/map/sources/soho.py:
```diff
@@ -41,7 +41,23 @@ def __init__(self, data, header, **kwargs):
+        """
+        Override the default implementation of coordinate_system to handle EITMAP-specific logic for CTYPE values.
+        """
+        ctype1 = self.meta.get('ctype1', None)
+        ctype2 = self.meta.get('ctype2', None)
+
+        if ctype1.lower() in ("solar-x", "solar_x"):
+            ctype1 = 'HPLN-TAN'
+
+        if ctype2.lower() in ("solar-y", "solar_y"):
+            ctype2 = 'HPLT-TAN'
 
+        return SpatialPair(ctype1, ctype2)
```
```suggestion
        ctype1 = self.meta.get('ctype1', None)
        ctype2 = self.meta.get('ctype2', None)
        if ctype1.lower() in ("solar-x", "solar_x"):
            ctype1 = 'HPLN-TAN'
        if ctype2.lower() in ("solar-y", "solar_y"):
            ctype2 = 'HPLT-TAN'
        return SpatialPair(ctype1, ctype2)
```

nabobalis commented on sunpy/map/sources/soho.py:
```diff
@@ -105,13 +121,25 @@ class LASCOMap(GenericMap):
+        """
+        Override the default implementation to handle LASCOMAP-specific logic for CTYPE values.
+        """
+        ctype1 = self.meta.get('ctype1', None)
```
What happens if meta is none? Would the if line not error?

tanishy7777 replied:
yea it would. WIll add a if self.meta condition for the same

nabobalis commented on sunpy/map/sources/tests/test_overriden_method.py:
```diff
@@ -0,0 +1,27 @@
+from sunpy.map.mapbase import SpatialPair
+from sunpy.data.test import get_dummy_map_from_header, get_test_filepath
+
+def test_eitmap_coordinate_system():
+    eit_map = sunpy.map.Map(get_test_filepath("EIT/efz20040301.000010_s.fits"))
+    assert eit_map.coordinate_system ==  SpatialPair(axis1='HPLN-TAN', axis2='HPLT-TAN')
+
+
+def test_lasco_coordinate_system():
+    lasco_map = get_dummy_map_from_header(get_test_filepath("lasco_c2_25299383_s.header"))
+    assert lasco_map.coordinate_system ==  SpatialPair(axis1='HPLN-TAN', axis2='HPLT-TAN')
+
+def test_xrt_coordinate_system():
+    xrt_map = get_dummy_map_from_header(get_test_filepath("HinodeXRT.header"))
+    assert xrt_map.coordinate_system ==  SpatialPair(axis1='HPLN-TAN', axis2='HPLT-TAN')
+
+def test_trace_coordinate_system():
+    trace_map = get_dummy_map_from_header(get_test_filepath("tsi20010130_025823_a2.header"))
+    assert trace_map.coordinate_system ==  SpatialPair(axis1='HPLN-TAN', axis2='HPLT-TAN')
+
+def test_sot_coordinate_system():
+    sot_map = get_dummy_map_from_header(get_test_filepath("HinodeSOT.header"))
+    assert sot_map.coordinate_system ==  SpatialPair(axis1='HPLN-TAN', axis2='HPLT-TAN')
+
```
These should go into the correct file for each source.

tanishy7777 replied:
got it

nabobalis commented:
The pre-commit check will also need to be fixed.

tanishy7777 commented:
pre-commit.ci autofix

tanishy7777 commented:
pre-commit.ci autofix

tanishy7777 commented:
@nabobalis I have made the changes. Please have a look at it

nabobalis commented on sunpy/map/sources/hinode.py:
```diff
@@ -60,6 +60,20 @@ def __init__(self, data, header, **kwargs):
+        """
+        Override the default implementation to handle SOTMap-specific logic for CTYPE values.
+        """
+        if self.meta:
```
What is this for?

tanishy7777 replied:
https://github.com/sunpy/sunpy/pull/7955#discussion_r1899842589
For this comment

tanishy7777 replied:
If self.meta is None the coordinate_system method wont be overriden

nabobalis replied:
Ah I misspoke. It might be empty but never none.

I mean to say what if `self.meta.get('ctype1', None)` is None. How does the line below it work?

tanishy7777 replied:
> I mean to say what if `self.meta.get('ctype1', None)` is None. How does the line below it work?
Ah ic. The way it was handled in `mapbase.py` was 

```python
ctype1 = self.meta.get('ctype1', None)
if not ctype1:
    warn_metadata("Missing CTYPE1 from metadata, assuming CTYPE1 is HPLN-TAN")
    ctype1 = 'HPLN-TAN'

ctype2 = self.meta.get('ctype2', None)
if not ctype2:
    warn_metadata("Missing CTYPE2 from metadata, assuming CTYPE2 is HPLT-TAN")
    ctype2 = 'HPLT-TAN'
```


But fromhttps://github.com/sunpy/sunpy/pull/7955#discussion_r1899732479
I thought we should keep that logic only in mapbase and warnings(warn_metadata) shouldnt be added in the map sources

nabobalis replied:
This logic is different and should stay in mapbase.

tanishy7777 replied:
> I mean to say what if `self.meta.get('ctype1', None)` is None. How does the line below it work?

If ctype1 or ctype2 is None then calling .lower() method on it will cause AttributeError
![image](https://github.com/user-attachments/assets/f91e87aa-2b52-4357-8d97-a2737738a9ec)


```python
 if ctype1.lower() in ("solar-x", "solar_x"):
 ```
 
 I think we can replace None with an empty string instead in this case?

nabobalis replied:
Lets do that.

tanishy7777 replied:
> Lets do that.

Will do that in all of the overriden coordinate_system methods. Should I do the same for `mapbase.py` ?

nabobalis replied:
Why not.

tanishy7777 replied:
> Why not.

Great! I have made the changes.

nabobalis commented on sunpy/map/sources/soho.py:
```diff
@@ -105,14 +119,27 @@ class LASCOMap(GenericMap):
     ----------
     * `SOHO Mission Page <https://sohowww.nascom.nasa.gov/>`_
     """
-
```
Add this back

nabobalis commented on sunpy/map/sources/tests/test_eit_source.py:
```diff
@@ -23,6 +24,8 @@ def test_fitstoEIT(eit_map):
 
+def test_eitmap_coordinate_system(eit_map):
+    assert eit_map.coordinate_system ==  SpatialPair(axis1='HPLN-TAN', axis2='HPLT-TAN')
 
```
```suggestion


```

nabobalis commented on sunpy/map/sources/tests/test_eit_source.py:
```diff
@@ -23,6 +24,8 @@ def test_fitstoEIT(eit_map):
     """Tests the creation of EITMap using FITS."""
     assert isinstance(eit_map, EITMap)
 
```
```suggestion


```

nabobalis commented on sunpy/map/sources/tests/test_lasco_source.py:
```diff
@@ -35,6 +36,8 @@ def test_fitstoLASCO(lasco_map):
     """Tests the creation of LASCOMap using FITS."""
     assert isinstance(lasco_map, LASCOMap)
 
```
```suggestion


```

nabobalis commented on sunpy/map/sources/tests/test_lasco_source.py:
```diff
@@ -35,6 +36,8 @@ def test_fitstoLASCO(lasco_map):
 
+def test_lasco_coordinate_system(lasco_map):
+    assert lasco_map.coordinate_system ==  SpatialPair(axis1='HPLN-TAN', axis2='HPLT-TAN')
 
```
```suggestion


```

nabobalis commented on sunpy/map/sources/tests/test_sot_source.py:
```diff
@@ -18,6 +19,8 @@ def test_fitstoSOT(sot):
     """Tests the creation of SOTMap using FITS."""
     assert isinstance(sot, SOTMap)
 
```
```suggestion


```

nabobalis commented on sunpy/map/sources/tests/test_sot_source.py:
```diff
@@ -18,6 +19,8 @@ def test_fitstoSOT(sot):
 
+def test_sot_coordinate_system(sot):
+    assert sot.coordinate_system ==  SpatialPair(axis1='HPLN-TAN', axis2='HPLT-TAN')
 
```
```suggestion


```

nabobalis commented on changelog/7955.deprecation.rst:
```diff
@@ -0,0 +1 @@
+deprecated `ctype1="solar-x"/"solar_x"` and `ctype2="solar-y"/"solar_y"` in `sunpy.map.Mapbase.GenericMap.coordinate_system`
```
We will need to workshop this message.

tanishy7777 replied:
> We will need to workshop this message.

I am sorry but I didn't get what you mean.

tanishy7777 replied:
> We will need to workshop this message.

```git
The use of `ctype1="solar-x"/"solar_x"` and `ctype2="solar-y"/"solar_y"` in 
`sunpy.map.Mapbase.GenericMap.coordinate_system` is deprecated and will 
be removed in a future release. Please update your code to use 
`ctype1="HPLN-TAN"` and `ctype2="HPLT-TAN"`.
```

How about this? @nabobalis

nabobalis replied:
Hmm, I think we will need other maintainers to chime in on this.

tanishy7777 replied:
> Hmm, I think we will need other maintainers to chime in on this.

works

nabobalis replied:
The unit test is failing in the oldest deps build, I am unsure why. That will need to be debugged.

tanishy7777 replied:
> The unit test is failing in the oldest deps build, I am unsure why. That will need to be debugged.

Will look into it.

nabobalis commented on sunpy/map/sources/tests/test_trace_source.py:
```diff
@@ -17,6 +18,8 @@ def test_fitstoTRACE(trace_map):
     """Tests the creation of TRACEMap using FITS."""
     assert isinstance(trace_map, TRACEMap)
 
```
```suggestion


```

nabobalis commented on sunpy/map/sources/tests/test_trace_source.py:
```diff
@@ -17,6 +18,8 @@ def test_fitstoTRACE(trace_map):
 
+def test_trace_coordinate_system(trace_map):
+    assert trace_map.coordinate_system ==  SpatialPair(axis1='HPLN-TAN', axis2='HPLT-TAN')
 
```
```suggestion


```

nabobalis commented on sunpy/map/sources/tests/test_xrt_source.py:
```diff
@@ -23,6 +24,8 @@ def test_fits_to_xrt(xrt_map):
     """Tests the creation of XRTMap using FITS."""
     assert isinstance(xrt_map, XRTMap)
 
```
```suggestion


```

nabobalis commented on sunpy/map/sources/tests/test_xrt_source.py:
```diff
@@ -23,6 +24,8 @@ def test_fits_to_xrt(xrt_map):
 
+def test_xrt_coordinate_system(xrt_map):
+    assert xrt_map.coordinate_system ==  SpatialPair(axis1='HPLN-TAN', axis2='HPLT-TAN')
 
```
```suggestion


```

nabobalis commented on sunpy/map/tests/test_mapbase.py:
```diff
@@ -351,6 +351,13 @@ def test_default_coordinate_system(generic_map):
     with pytest.warns(SunpyMetadataWarning, match='Missing CTYPE2 from metadata'):
         assert generic_map.coordinate_system == ('HPLN-TAN', 'HPLT-TAN')
 
```
```suggestion


```

tanishy7777 commented:
pre-commit.ci autofix

tanishy7777 commented:
pre-commit.ci autofix

nabobalis commented on sunpy/map/mapbase.py:
```diff
@@ -1233,9 +1234,11 @@ def coordinate_system(self):
         # changes it to a ctype that is understood.  See Thompson, 2006, A.&A.,
         # 449, 791.
         if ctype1.lower() in ("solar-x", "solar_x"):
+            warnings.warn("CTYPE1 value 'solar-x'/'solar_x' is deprecated, use 'HPLN-TAN' instead.", SunpyDeprecationWarning)
```
Why was this replaced?

tanishy7777 replied:
I thought https://github.com/sunpy/sunpy/pull/7955#discussion_r1900340785
might be due to the warn_deprecated method so I tried using warnings module instead but that dosent seem to be the issue.

tanishy7777 replied:
Is there a way to run github-actions locally, to avoid polluting the PR? @nabobalis

nabobalis replied:
You can run each of the test environments locally: https://docs.sunpy.org/en/latest/dev_guide/contents/tests.html

nabobalis replied:
Can you restore the previous warnings handling?

tanishy7777 replied:
> Can you restore the previous warnings handling?

done

nabobalis commented on changelog/7955.deprecation.rst:
```diff
@@ -0,0 +1 @@
+deprecated `ctype1="solar-x"/"solar_x"` and `ctype2="solar-y"/"solar_y"` in `sunpy.map.Mapbase.GenericMap.coordinate_system`
```
```suggestion
Deprecated support for phrasing CTYPE SOLAR-X and SOLAR-Y within `sunpy.map.Mapbase.GenericMap`.
These are not supported by the FITS standard and are instead to be handled within a map source.
```
Maybe?

ayshih commented on sunpy/map/mapbase.py:
```diff
@@ -1218,12 +1219,12 @@ def coordinate_system(self):
         If not present, defaults to (HPLN-TAN, HPLT-TAN), and emits a warning.
         """
-        ctype1 = self.meta.get('ctype1', None)
+        ctype1 = self.meta.get('ctype1', '')
```
What is the purpose of this change?

tanishy7777 replied:
In response to https://github.com/sunpy/sunpy/pull/7955#discussion_r1900196416 and https://github.com/sunpy/sunpy/pull/7955#discussion_r1900334917

I replaced None with '' to avoid AttributeError when `.lower()` is called on `None`

ayshih replied:
Unless I'm missing something I'd revert this change and the similar one below it.  No need to change lines that don't need to be changed.

tanishy7777 replied:
> Unless I'm missing something I'd revert this change and the similar one below it. No need to change lines that don't need to be changed.

yep that is true will make it `None` again

ayshih reviewed (changes requested):
The deprecation warnings and changelog entry should be phrased along the lines of "General support within `~sunpy.map.GenericMap` for the non-standard CTYPE value 'SOLAR-X' is now deprecated, and should instead be handled by a source-specific subclass."

ayshih commented on sunpy/map/sources/soho.py:
```diff
@@ -113,6 +127,20 @@ def __init__(self, data, header, **kwargs):
+        """
+        Override the default implementation to handle LASCOMAP-specific logic for CTYPE values.
+        """
+        if self.meta:
+            ctype1 = self.meta.get('ctype1', '')
+            ctype2 = self.meta.get('ctype2', '')
+            if ctype1.lower() in ("solar-x", "solar_x"):
+                ctype1 = 'HPLN-TAN'
+            if ctype2.lower() in ("solar-y", "solar_y"):
+                ctype2 = 'HPLT-TAN'
+            return SpatialPair(ctype1, ctype2)
```
```suggestion
        ctype1, ctype2 = self.meta['ctype1'], self.meta['ctype2']
        if ctype1.lower() in ("solar-x", "solar_x"):
            ctype1 = 'HPLN-TAN'
        if ctype2.lower() in ("solar-y", "solar_y"):
            ctype2 = 'HPLT-TAN'
        return SpatialPair(ctype1, ctype2)
```
Same here: this overridden method ought to error if `meta` does not have `ctype1` or `ctype2`.

ayshih commented on sunpy/map/sources/trace.py:
```diff
@@ -55,6 +55,20 @@ def __init__(self, data, header, **kwargs):
+        """
+        Override the default implementation to handle TRACEMAP-specific logic for CTYPE values.
+        """
+        if self.meta:
+            ctype1 = self.meta.get('ctype1', '')
+            ctype2 = self.meta.get('ctype2', '')
+            if ctype1.lower() in ("solar-x", "solar_x"):
+                ctype1 = 'HPLN-TAN'
+            if ctype2.lower() in ("solar-y", "solar_y"):
+                ctype2 = 'HPLT-TAN'
+            return SpatialPair(ctype1, ctype2)
```
```suggestion
        ctype1, ctype2 = self.meta['ctype1'], self.meta['ctype2']
        if ctype1.lower() in ("solar-x", "solar_x"):
            ctype1 = 'HPLN-TAN'
        if ctype2.lower() in ("solar-y", "solar_y"):
            ctype2 = 'HPLT-TAN'
        return SpatialPair(ctype1, ctype2)
```
Same here: this overridden method ought to error if `meta` does not have `ctype1` or `ctype2`.

ayshih reviewed (changes requested):
Almost there!  The checking and fallbacks in the source-specific subclasses shouldn't exist.  We want those to error if the header does not conform.

ayshih commented:
Er, do we know why the `oldestdeps` CI run is failing?

tanishy7777 commented:
> Er, do we know why the `oldestdeps` CI run is failing?

to test that `mapbase.py` was giving deprecation warnings I had added the following code in `test_mapbase.py`

```python
  def test_coordinate_system_solar_x_solar_y(generic_map):
        generic_map.meta['ctype1'] = 'SOLAR-X'
        generic_map.meta['ctype2'] = 'SOLAR-Y'
       with pytest.warns(SunpyDeprecationWarning, match="CTYPE1 value 'solar-x'/'solar_x' is deprecated") 
```

but for some reason the `SunpyDeprecationWarning` warning is not generated on 
![image](https://github.com/user-attachments/assets/27406f15-dc0d-4fb2-a4e2-e95b2489cdf2)
oldestdeps

nabobalis commented:
Should be fixed now.

tanishy7777 commented:
yep now its working

nabobalis commented:
> yep now its working. Why did you remove the request for review? @nabobalis

Because only the two humans requested are needed.

tanishy7777 commented:
> Because only the two humans requested are needed.

oh ic thx

ayshih reviewed (approved):
Looks good to me.  Thanks for your patience!

nabobalis commented:
Thanks again @tanishy7777
